pkg/apis/pipeline/v1beta1/task_validation_test.go

```diff
@@ -855,6 +873,50 @@ func TestTaskSpecValidateError(t *testing.T) {
 			Message: `variable is not properly isolated in "not isolated: $(params.baz[*])"`,
 			Paths:   []string{"steps[0].args[0]"},
 		},
```

Comment:
Maybe add another test like this for object?

Reply:
Just added a test named `object star used in unaccepted field (not isolated)`. 

But I think this might be a bit duplicated with the [existing two tests](https://github.com/tektoncd/pipeline/pull/4861/files#diff-624ee2137fe5ef32422ce97821c958d84a30127121ae8a9183257ed85f417d03R877-R919) that I added before. Unlike array case, using the whole object param will be invalid no matter if it's isolated according to this [tep](https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md#variable-replacement-with-object-params). 
> When populating a string field, It would be invalid (at least initially) to attempt to do variable replacement on the entire gitrepo object ($(params.gitrepo)). If we choose to support this later we could potentially replace the value with the json representation of the object directly.


Let me know what you think. Happy to remove or keep.

Reply:
Ah I meant to say putting it in a valid field (like a object type of params) but not isolated

Reply:
> Ah I meant to say putting it in a valid field (like a object type of params) but not isolated

I'm still confused. Do you mean using object param in a valid field which can accept object as whole?

Reply:
> Ah I meant to say putting it in a valid field (like a object type of params) but not isolated

I'm still confused. Do you mean using object param in a valid field which can accept object as whole?